File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
from heapq import *
def solution(scoville, K):
    answer = 0
    heapify(scoville)
    if scoville[0] >= K:
        return 0
    while len(scoville) > 1:
        f = heappop(scoville)
        s = heappop(scoville)
        heappush(scoville, f+s*2)
        answer += 1
        if scoville[0] >= K:
            break
    if scoville[0] < K:
        answer = -1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for(int s : scoville){
            pq.add(s);
        }
        
        int answer = 0;
        while(pq.size() >= 2 && pq.peek() < K){
            int a = pq.poll();
            int b = pq.poll();
            pq.add(a + b*2);
            answer++;
        }
        
        return pq.peek() < K ? -1 : answer;
    }
}